ddtrace/contrib/boto/patch.py

```diff
@@ -41,12 +49,16 @@ def patch():
     wrapt.wrap_function_wrapper(
         "boto.connection", "AWSAuthConnection.make_request", patched_auth_request
     )
-    Pin(service="aws", app="aws", app_type="web").onto(
-        boto.connection.AWSQueryConnection
-    )
-    Pin(service="aws", app="aws", app_type="web").onto(
-        boto.connection.AWSAuthConnection
-    )
+    Pin(
+        service=config.aws['service_name'],
+        app="aws",
+        app_type="web",
+    ).onto(boto.connection.AWSQueryConnection)
+    Pin(
+        service=config.aws['service_name'],
+        app="aws",
+        app_type="web",
+    ).onto(boto.connection.AWSAuthConnection)
```

Comment:
Notice that the 'aws' service name will overlap with botocore. It's OK for pinning, but for the config object, perhaps this might cause unexpected behavior if people are configuring boto, patching it, and then configuring botocore and patching it in a separate module/import flow.

Simplest thing to do here is just put these in boto/botocore in the config object.